src/modules/menu/components/menu/MenuTray.vue

```diff
@@ -1,29 +1,116 @@
+<script setup>
+import { computed, onMounted, ref, toRefs, watch } from 'vue'
+import { useI18n } from 'vue-i18n'
+import { useStore } from 'vuex'
+
+import MenuThreeD from '@/modules/menu/components/3d/MenuThreeD.vue'
+import MenuActiveLayersList from '@/modules/menu/components/activeLayers/MenuActiveLayersList.vue'
+import MenuAdvancedToolsList from '@/modules/menu/components/advancedTools/MenuAdvancedToolsList.vue'
+import MenuSection from '@/modules/menu/components/menu/MenuSection.vue'
+import MenuPrintSection from '@/modules/menu/components/print/MenuPrintSection.vue'
+import MenuSettings from '@/modules/menu/components/settings/MenuSettings.vue'
+import MenuShareSection from '@/modules/menu/components/share/MenuShareSection.vue'
+import MenuTopicSection from '@/modules/menu/components/topics/MenuTopicSection.vue'
+
+const dispatcher = { dispatcher: 'MenuTray.vue' }
+
+const i18n = useI18n()
+const store = useStore()
+
+const props = defineProps({
+    compact: {
+        type: Boolean,
+        default: false,
+    },
+})
+const { compact } = toRefs(props)
+
+const refs = ref({})
+// multiMenuSections means that they can be open together
+const multiMenuSections = ref(['topicsSection', 'activeLayersSection'])
+// singleModeSections means that those section cannot be open together with other
+// sections and would therefore toggle other sections automatically.
+const singleModeSections = ref([
+    'drawSection',
+    'settingsSection',
+    'shareSection',
+    'toolsSection',
+    'printSection',
+    '3dSection',
+])
+
+const is3dMode = computed(() => store.state.cesium.active)
+const showImportFile = computed(() => store.state.ui.importFile)
+const showDrawingOverlay = computed(() => store.state.drawing.drawingOverlay.show)
+const mapModuleReady = computed(() => store.state.app.isMapReady)
+
+watch(showImportFile, (show) => {
+    if (show) {
+        refs.value.activeLayersSection.open()
+    }
+})
+
+onMounted(() => {
+    if (is3dMode.value) {
+        refs.value['3dSection'].open()
+    }
+})
+
+function toggleDrawingOverlay() {
+    store.dispatch('toggleDrawingOverlay', dispatcher)
+}
+
+function onOpenMenuSection(id) {
+    let toClose = singleModeSections.value.filter((section) => section !== id)
+    if (singleModeSections.value.includes(id)) {
+        toClose = toClose.concat(multiMenuSections.value)
+    }
+    toClose.forEach((section) => refs.value[section]?.close())
+}
+
+function onCloseMenuSection(id) {
+    if (['drawSection', 'toolsSection'].includes(id)) {
+        refs.value.activeLayersSection.open()
+    }
+}
+
+function updateRef(el) {
```

Comment:
Maybe this could benefit of a littledocumentation as it's not directly clear what it's doing